src/Debugger-Model/TDebugger.trait.st

```diff
@@ -34,6 +34,11 @@ TDebugger classSide >> availableAutomatically: anObject [
 	availableAutomatically := anObject
 ]
 
+{ #category : #accessing }
+TDebugger classSide >> closeAllDebuggers [
+	^ self explicitRequirement
+]
```

Comment:
I agree with the changes.

However, why not having the default behavior (do nothing) in TDebugger instead of an explicit requirement, then debugger classes redefine it to implement their own logic?

I am asking genuinely, I do not know about what would be the best solution.
I would say, generally, provide a default behavior and avoid redefining the same default behavior everywhere, but I understand that the idea is to force all debuggers to reimplement that method. 

In the other hand, there are nothing constraining debuggers to close all of their instances, it is just a practicality offered to developers.

Reply:
The idea is indeed to force all debuggers to reimplement that method.
Having it as an explicit requirement means that an explicit error will pop up the next time someone (like Pablo here) clicks "close all debuggers" and it doesn't work. So the user will be told what the error is (missing implementation of #closeAllDebuggers) instead of the "close all debuggers" not working silently.

Reply:
I know what it does, but what I mean is that it looks like a code smell.

Ensure that all debuggers to which we ask "close all debuggers" can answer => **yes**
Duplicating the same empty method in all debuggers that do not need it redefined => **smell**
Having a default empty method in the Trait like I suggested => **smell**

How can we do it better?
Maybe "close all debuggers" is the wrong naming, and if we find the naming corresponding to the exact behavior we are looking for, it will be easier to find a solution.

Maybe this method do not belong to the Trait, and should be in the debugger system itself?
It seems odd that, if you ask "close all dbg" to GT, it closes only the GT debuggers **windows**... maybe that should be a "system-level" method that closes/terminates all windows/presenters/models/instances of all TDebugger users?

Reply:
```Maybe this method do not belong to the Trait, and should be in the debugger system itself?
It seems odd that, if you ask "close all dbg" to GT, it closes only the GT debuggers windows... maybe that should be a "system-level" method that closes/terminates all windows/presenters/models/instances of all TDebugger users?```

Ok, my bad, the "close all debuggers" method is already a system-level method.
So from there, we need a way to avoid the smells above.

Reply:
Hello guys what is the next step?
Should I integrate this PR?
Tx

Reply:
It's an open discussion. We could say that we integrate it as is so that it fixes the problem, and take care of the problem in a second time.

Reply:
Yes, doing things step by step is good